File: csharp/ChainOfResponsibility-Pets/Hamster.cs
public class Hamster : Pet
{
    public override string HandleTrap(Bait bait)
    {
        if (bait == Bait.POPCORN)
            return "happy hamster";
        else
            return nextPet.HandleTrap(bait);
    }

    public Hamster(Pet nextPet) : base(nextPet)
    {
    }
}

File: java/src/main/java/codingdojo/Hamster.java
package codingdojo;

public class Hamster extends Pet {
    public Hamster(Pet nextPet) {
        super(nextPet);
    }

    @Override
    public String handleTrap(Bait bait) {
        if (bait == Bait.POPCORN)
            return "happy hamster";
        else
            return nextPet.handleTrap(bait);
    }
}
